models/pagination.go

```diff
@@ -0,0 +1,56 @@
+package models
+
+import "strconv"
+
+type Filter struct {
+	Name, Value string
+}
+
+type Pagination struct {
+	limit   int
+	offset  int
+	filters []Filter
+}
+
+func NewPagination(filters ...Filter) *Pagination {
+	return &Pagination{
+		filters: filters,
+	}
+}
+
+func (p *Pagination) Limit() int {
+	return p.limit
+}
+
+func (p *Pagination) Offset() int {
+	return p.offset
+}
+
+func (p *Pagination) Filter(name string) (Filter, bool) {
+	for _, f := range p.filters {
+		if f.Name == name {
+			return f, true
+		}
+	}
+	return Filter{}, false
+}
+
+func (p *Pagination) ToPairs() []string {
```

Comment:
Opted for a ToPairs method that converts everything with a non-default (valid) value to a pairs string array that can be passed on in a `PathTo` or `URLTo` method.

Reply:
Good idea, now I understand your question about url.Values